### PR TITLE
Feature/dina PR

### PR DESCRIPTION
**Pull request by dinboyko**
Major changes:

- io.py: Eliminated library pyqtgraph for the prefix unit conversion. The function was implemented in util.py

- All scripts: Modifies documentation and formatting to get a higher flake8(PEP standart) and pylint score.

- Test files for scripts in src/ioniq

- GitHub Actions: modified pylint.yml to make it go through all the files regardless the error, added pytest.yml 

PS: I will keep the next PRs smaller

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # first case: standard split
+        (np.array([0, 0, 1, 1, 2, 2]), 0, False, (np.array([0, 2, 4]), np.array([2, 4, 6]))),
+        # n_remove = 1
+        (np.array([0, 0, 1, 1, 2, 2]), 1, False, (np.array([1, 3, 5]), np.array([2, 4, 6]))),
+        # as_tuples=True
+        (np.array([0, 0, 1, 1, 2, 2]), 0, True, [(0, 2), (2, 4), (4, 6)]),
+        # case with empty array
+        (np.array([]), 0, False, ([], []))
+    ]
+)
```
Comment (a-fa) [inline tests/test_utils.py:43]: to be discussed, the formatting of the expected return values seems non-intuitive.

File: tests/testTrace.py
Comment (a-fa) [inline tests/testTrace.py:1]: how come are some tests in pytest and others in unittest?

File: .github/workflows/pytest.yml
Comment (a-fa) [inline .github/workflows/pytest.yml:1]: add windows-latest, macos-latest to runs-on 
https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/choosing-the-runner-for-a-job

File: tests/testTrace.py
Comment (dinboyko) [reply tests/testTrace.py:1]: three test files: testDummy, testGuiBasic and testTrace are from PythIon.

**Review by a-fa — APPROVED** [review]

**Pull request merged**